tests/db_access/test_utils.hpp

```diff
@@ -7,7 +7,34 @@
 
 #include <chrono>
 #include <thread>
+#include <iostream>
+#include <streambuf>
+#include <string>
+#include <sstream>
 
+template <typename T>
+class StreamRedirect {
+    public:
+        StreamRedirect(T &stream): buffer{}, originalStream{stream},
+            originalBuffer{stream.rdbuf(buffer.rdbuf())} {}
+
+        std::string getString() {
+            return buffer.str();
+        }
+
+        void restore() {
+            originalStream.rdbuf(originalBuffer);
+        }
+
+        ~StreamRedirect() {
+            restore();
+        }
+
+    private:
+        std::stringstream buffer;
+        T &originalStream;
+        std::streambuf *originalBuffer;
```

Comment:
We can simplify the logic by not storing `originalBuffer`, as each istream already can retrieve a pointer to its internal buffer.
See https://stackoverflow.com/questions/49591500/difference-between-internal-vs-associated-stream-buffer

Note that if the passed stream would have associated buffer different than internal one, this simplification would not work, but we may not care about this case.

Reply:
We decided that it isn't worth the change, instead enhance commentary of current solution.